src/test/scala/nl.knaw.dans.easy.deposit/docs/DatasetMetadataSpec.scala

```diff
@@ -129,85 +140,142 @@ class DatasetMetadataSpec extends TestSupportFixture {
     val s: JsonInput =
       """{
         |  "relations": [
-        |    { "qualifier": "dcterms:hasFormat", "url": "string" },
+        |    { "qualifier": "dcterms:hasFormat", "url": "http://host" },
         |    { "value": "string", "qualifier": "dcterms:hasFormat" }
         |  ]
         |}""".stripMargin
     DatasetMetadata(s) shouldBe a[Success[_]]
   }
 
   it should "accept incomplete relations in a different order" in {
-    val s: JsonInput =
+    val s =
       """{
         |  "relations": [
         |    { "value": "string", "qualifier": "dcterms:hasFormat" },
-        |    { "qualifier": "dcterms:hasFormat", "url": "string" }
+        |    { "qualifier": "dcterms:hasFormat", "url": "http://host" }
         |  ]
         |}""".stripMargin
     DatasetMetadata(s) shouldBe a[Success[_]]
   }
 
   it should "accept a RelatedIdentifier without scheme" in {
-    val s: JsonInput = """{ "relations": [ { "value": "string", "qualifier": "dcterms:hasFormat" } ] }"""
-    DatasetMetadata(s) shouldBe a[Success[_]]
+    DatasetMetadata("""{ "relations": [ { "value": "string", "qualifier": "dcterms:hasFormat" } ] }""") shouldBe a[Success[_]]
   }
 
   it should "accept a Relation without title" in {
-    val s: JsonInput = """{ "relations": [ { "qualifier": "dcterms:hasFormat", "url": "string" } ] }"""
-    DatasetMetadata(s) shouldBe a[Success[_]]
+    DatasetMetadata("""{ "relations": [ { "qualifier": "dcterms:hasFormat", "url": "http://host" } ] }""") shouldBe a[Success[_]]
   }
 
   it should "accept a Relation without url" in {
-    val s: JsonInput = """{ "relations": [ { "qualifier": "dcterms:hasFormat", "title": "string" } ] }"""
-    DatasetMetadata(s) shouldBe a[Success[_]]
+    DatasetMetadata("""{ "relations": [ { "qualifier": "dcterms:hasFormat", "title": "string" } ] }""") shouldBe a[Success[_]]
   }
 
   it should "reject a relation with just a qualifier" in {
-    val s: JsonInput = """{ "relations": [ { "qualifier": "dcterms:hasFormat" } ] }"""
-    shouldReturnCustomMessage(s, """expected one of (Relation | RelatedIdentifier) got: {"qualifier":"dcterms:hasFormat"}""")
+    """{ "relations": [ { "qualifier": "dcterms:hasFormat" } ] }"""
+      .causesInvalidDocumentException( """expected one of (Relation | RelatedIdentifier) got: {"qualifier":"dcterms:hasFormat"}""")
+  }
```

Comment:
**Too big for the scope of the pull request**
Shouldn't we keep the barbarians at the gate and somehow escape strings that are vulnerable to injection? The following unit test shows just one simple case that now gets through. 

```suggestion
  }
  
    private val valueWithHtmlInjection = """<a href="javascript:alert('hello')">Foo<a> bar"""
  "values with injected html" should behave like validDatasetMetadata(
    input = Try(new MinimalDatasetMetadata(alternativeTitles = Some(Seq(valueWithHtmlInjection)))),
    subset = actualDDM => dcmiMetadata(actualDDM),
    expectedDdmContent =
      <ddm:dcmiMetadata>
        <dcterms:identifier xsi:type="id-type:DOI">mocked-DOI</dcterms:identifier>
        <dcterms:alternative>{ valueWithHtmlInjection }</dcterms:alternative>
        <dcterms:dateSubmitted xsi:type="dcterms:W3CDTF">2018-03-22</dcterms:dateSubmitted>
      </ddm:dcmiMetadata>
  )
```